libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -449,6 +455,35 @@ constructor(
         tripSession.unregisterStateObserver(tripSessionStateObserver)
     }
 
+    /**
+     * Attach your own controller to determine when drivers arrived at stops via [ArrivalController]
+     * Use [navigateToNextStop] to manually move navigator to the next stop.
+     *
+     * @param arrivalController ArrivalObserver
+     */
+    fun attachArrivalController(arrivalController: ArrivalController) {
+        arrivalProgressObserver.attach(arrivalController)
+    }
+
+    /**
+     * Stop controlling arrival at stops, detaching will require the caller
+     * to call [navigateToNextStop] manually.
+     */
+    fun removeArrivalController() {
+        tripSession.unregisterRouteProgressObserver(arrivalProgressObserver)
+        arrivalProgressObserver.detach()
+    }
+
+    /**
+     * After arriving a stop, this can be used to manually decide when to start
+     * navigating to the next stop. Use the [ArrivalController] to require manual navigation.
+     *
+     * @return true if navigation to next stop could be started, false otherwise
+     */
+    fun navigateToNextStop(): Boolean {
```

Comment:
What happens if this is called before arriving at a stop and multiple times? Are we checking somehow that this API is not misused?

Reply:
yeah the check for misuse is in the return value. i wasn't going to introduce crashes until this works as expected.

right now, it doesn't seem to be navigating to next stop

Reply:
Could it be because downstream in NN they have a check 👀 for `ROUTE_ARRIVED` event so you can only `navigateToNextStop` safely? If so, we don't need to add any extra check on our side 🚀 Is that the case @mskurydin?

Reply:
seems like if you call this too early you'll get an "off-route" event..

to the, what if you call it multiple times, your next calls could be dropped. the Boolean return is the best check for misuse right now